src/main/java/world/anhgelus/molehunt/Game.java

```diff
@@ -34,11 +34,20 @@ public Game(MinecraftServer server) {
     }
 
     public void start() {
-        if (started) return;
-        final int n = (server.getCurrentPlayerCount() - server.getCurrentPlayerCount() % 4)/4;
+        final int n;
+        if (Molehunt.CONFIG.MOLE_COUNT < 0) {
+            n = Math.floorDiv(server.getCurrentPlayerCount(), Math.floorDiv(100, (int) Molehunt.CONFIG.MOLE_PERCENTAGE));
+        } else {
+            n = Molehunt.CONFIG.MOLE_COUNT;
+        }
+
         final var playerManager = server.getPlayerManager();
-        final var players = new ArrayList<>(playerManager.getPlayerList());
+
+        final var players = playerManager.getPlayerList();
         for (int i = 0; i < n; i++) {
+            // Can happen if the mole count is greater than the player count
+            if (players.isEmpty()) break;

```

Comment:
Do not use a break. Add this condition inside the for:
```java
for (int i = 0; i < n && !players.isEmpty(); i++) {
```

Reply:
I find this more readable, but I'll change it